transportable-udfs-utils/src/main/java/com/linkedin/transport/utils/FileSystemUtils.java

```diff
@@ -44,17 +45,17 @@ public static boolean isLocalEnvironment(Configuration conf) {
    *
    * @return the HDFS FileSystem if we are not in local mode, local FileSystem if we are.
    */
-  public static FileSystem getHDFSFileSystem() {
+  public static FileSystem getHDFSFileSystem(String filePath) {
     FileSystem fs;
     JobConf conf = new JobConf();
     try {
       // Checks if currently we are in local mode, which is basically when running unit tests
       if (isLocalEnvironment(conf)) {
         fs = FileSystem.getLocal(conf);
       } else {
-        fs = FileSystem.get(conf);
+        fs = FileSystem.get(new java.net.URI(filePath), conf);
```

Comment:
Prefer Path.getFileSystem(conf) since the Path constructor does some handling for normalization and unescaped characters.